common/models/templates/src/common.ts

```diff
@@ -152,3 +152,21 @@ export function defaultApplyCasing(casing: CasingForm, text: string): string {
       return text.substring(0, headUnitLength).toUpperCase() .concat(text.substring(headUnitLength));
   }
 }
+
+/**
+ * An **opaque** type for a string that is exclusively used as a search key in
+ * the trie. There should be a function that converts arbitrary strings
+ * (queries) and converts them into a standard search key for a given language
+ * model.
+ *
+ * Fun fact: This opaque type has ALREADY saved my bacon and found a bug!
```

Comment:
```suggestion
```

I realise this isn't a new comment, but this kind of comment doesn't need to stay in the codebase. Put it in Slack instead :grin: